test/test_buffer.cpp

```diff
@@ -139,6 +139,11 @@ BOOST_AUTO_TEST_CASE(destructor_templated_callback)
         boost::compute::buffer buf(context, 128);
         buf.set_destructor_callback(boost::bind(destructor_templated_callback_function, &invoked));
     }
+
+    // NVIDIA calls the callback in a background thread and so the check is executed
+    // before the callback function is called
+    sleep(1);
```

Comment:
Does this always work for you? It seems somewhat fragile to me. Ideally this could be implemented without a sleep of some arbitrary amount of time.